docs/Windows-VisualStudio.md

```diff
@@ -100,7 +103,8 @@ There is also a skeleton game project created for you at https://github.com/32bl
       "inheritEnvironments": [ "gcc-arm" ],
       "variables": [],
       "intelliSenseMode": "linux-gcc-arm",
-      "cmakeToolchain": "[path...]/32blit-sdk/32blit.toolchain"
+      "cmakeToolchain": "[path...]/32blit-sdk/32blit.toolchain",
+      "environments": [ { "PATH": "${env.PATH}" } ]
```

Comment:
While that does fix the build, it usually breaks configuring as it resets the PATH set by the "gcc-arm" environment containing the cross tools... (Works after configuring as all the paths are cached)

Reply:
I wonder if it can do something like append to the path rather than completely trounce it. I assume you've probably bashed your head against this quite a bit! :grimacing:

Reply:
That said, I can't seem to get it to break configuration. Even after deleting the cache.

Reply:
Yeah I kept trying to workaround it in `32blit setup`... and eventually got annoyed enough to remember my MS account and report it.

Reply:
> That said, I can't seem to get it to break configuration. Even after deleting the cache.

Possibly finding some other installed toolchain instead of the VS supplied one? It definitely doesn't find the one shipped with VS with that override.

Reply:
Oooh that's something to check. Might be prudent to remove the ARM GCC toolchain since that's in my path.

I strongly suspect there wont be a fix for this imminently and we'd have to bless the command-line procedure for actual on-device builds. Which isn't so bad.

Or, I guess, if it's gonna detect ARM GCC and work... that's also fine?

Reply:
Hah, maybe it'll work in VS 2022, who knows... There's always VS Code (and probably most other editors) to make the "command-line" builds less... command-line.

Guess the problem with an "external" toolchain is it might stop working, as finding that instead of the VS one probably isn't supposed to work. Might mess up intellisense or something :shrug:

Reply:
You're right- it finds the ARM GCC toolchain I installed. Makes sense!

This... sort-of... fix does the trick in both cases:

```json
"environments": [ { "PATH": "${env.ProgramFiles(x86)}/Microsoft Visual Studio/2019/Community/Linux/gcc_arm/bin/;${env.PATH}" } ]
```

It's ugly, and I'd feel a lot happier if I could find a way to get the *full* path to the currently running Visual Studio but there doesn't seem to be a way. Also not sure if I'm subtly breaking something by supplying the wrong executable, my install isn't exactly clean!

Reply:
In VS2022 it seems that the "IoT" configs are gone, so the only way to get this to work is to install the Arm toolchain and manually add a config similar to the one above. (Including the PATH hack...)

Anyway, I was in the area so I've rebased this.